### PR TITLE
Allow user tune KafkaConfig

### PR DESCRIPTION
**Pull request by IgnisDeusZero**
This change allows users to customize Kafka clients that is used internally. 
For example enable debug logs or set up a security protocol

### CODE REVIEW

File: src/Vektonn.DataSource/Kafka/KafkaConsumerConfig.cs [hunk 1]
```diff
@@ -20,5 +21,17 @@ public KafkaConsumerConfig(string[] bootstrapServers)
         public TimeSpan MaxRetryDelay { get; set; } = TimeSpan.FromSeconds(10);
         public TimeSpan WatermarkOffsetsQueryTimeout { get; set; } = TimeSpan.FromSeconds(1);
         public int ConsumeBatchSize { get; set; } = 10_000;
+        public Action<ConsumerConfig> ConsumerConfigUserTuning { get; set; } = config => {};
```
Comment (kvadros) [inline src/Vektonn.DataSource/Kafka/KafkaConsumerConfig.cs:24]: Naming suggestions (for this property and method below):
* `UserTuning` looks like `Customisation`, `Tuning` looks like `Initialisation`;
* They should be verbs, not nouns (e.g. `CustomiseConsumerConfig` and `InitialiseConsumerConfig`).

**Review by kvadros — CHANGES_REQUESTED** [review]

File: src/Vektonn.DataSource/Kafka/KafkaConsumer.cs [hunk 2]
```diff
@@ -41,34 +41,11 @@ internal class KafkaConsumer : IDisposable
...
-                    (_, topicPartitions) =>
-                    {
-                        var offsetsToConsumeFrom = committedOffsets.GetOffsetsToConsumeFrom(topicPartitions);
-                        this.log.Info($"Assigned {topicPartitions.Count} partitions with offsetsToConsumeFrom: {string.Join("; ", offsetsToConsumeFrom.Select(x => x.ToString()))}");
-                        firstAssignmentSignal.Set();
-                        return offsetsToConsumeFrom;
-                    });
+            var consumerConfig = new ConsumerConfig();
+            kafkaConsumerConfig.ConsumerConfigUserTuning(consumerConfig);
+            kafkaConsumerConfig.ConsumerConfigTuning(consumerConfig);
```
[the remark this answers is not in the document]
Comment (IgnisDeusZero) [reply src/Vektonn.DataSource/Kafka/KafkaConsumer.cs]: https://github.com/vektonn/vektonn/pull/13/commits/520265f1b6939e76f9dbc612e4fa4394e9776dd8
Renamed methods and delegates

File: src/Vektonn.DataSource/Kafka/KafkaConsumerConfig.cs [hunk 3]
```diff
@@ -20,5 +21,17 @@ public KafkaConsumerConfig(string[] bootstrapServers)
...
+
+        public void ConsumerConfigTuning(ConsumerConfig consumerConfig)
```
[the remark this answers is not in the document]
Comment (IgnisDeusZero) [reply src/Vektonn.DataSource/Kafka/KafkaConsumerConfig.cs]: https://github.com/vektonn/vektonn/pull/13/commits/520265f1b6939e76f9dbc612e4fa4394e9776dd8
Change modifier to internal

File: src/Vektonn.DataSource/Kafka/KafkaConsumer.cs [hunk 4]
```diff
@@ -97,6 +74,26 @@ public async Task RunAsync(CancellationToken cancellationToken)
             await initializationTcs.Task;
         }
 
+        private IConsumer<byte[], byte[]> BuildConsumer(ConsumerConfig consumerConfig)
+        {
+            var consumerBuilder = new ConsumerBuilder<byte[], byte[]>(consumerConfig)
```
[the remark this answers is not in the document]
Comment (IgnisDeusZero) [reply src/Vektonn.DataSource/Kafka/KafkaConsumer.cs]: https://github.com/vektonn/vektonn/pull/13/commits/2143dfab6181fe3bb1b9c4e2816bb00c7811bea4
done

File: src/Vektonn.DataSource/Kafka/KafkaConsumerConfig.cs (hunk 1 above)
Comment (IgnisDeusZero) [reply src/Vektonn.DataSource/Kafka/KafkaConsumerConfig.cs]: Thank you for suggestions! Changed names in conversation above

Comment (IgnisDeusZero) [reply src/Vektonn.DataSource/Kafka/KafkaConsumerConfig.cs]: https://github.com/vektonn/vektonn/commit/520265f1b6939e76f9dbc612e4fa4394e9776dd8
Renamed methods and delegates

File: src/Vektonn.DataSource/Kafka/KafkaConsumer.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (IgnisDeusZero) [reply src/Vektonn.DataSource/Kafka/KafkaConsumer.cs]: Internal customization mostly customize settings, that exist in Kafka(Consumer|Producer)Config. 
And hardcoded settings are important for correct work of dataSource

**Pull request merged**